src/vs/workbench/browser/parts/notifications/notificationsTelemetry.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable } from 'vs/base/common/lifecycle';
+import { INotificationService } from 'vs/platform/notification/common/notification';
+import { ITelemetryService } from 'vs/platform/telemetry/common/telemetry';
+import { IWorkbenchContribution } from 'vs/workbench/common/contributions';
+import { hash } from 'vs/base/common/hash';
+
+interface NotificationMetrics {
+	id: number;
+	source?: string;
+}
+
+type NotificationMetricsClassification = {
+	id: { classification: 'SystemMetaData', purpose: 'FeatureInsight' };
+	source?: { classification: 'SystemMetaData', purpose: 'FeatureInsight' }
+};
+
+export class NotificationsTelemetryContribution extends Disposable implements IWorkbenchContribution {
+
+	constructor(
+		@ITelemetryService private readonly telemetryService: ITelemetryService,
+		@INotificationService private readonly notificationService: INotificationService
+	) {
+		super();
+		this.registerListeners();
+	}
+
+	private registerListeners(): void {
+		this._register(this.notificationService.onDidAddNotification(notification => {
+			if (!notification.silent) {
+				this.telemetryService.publicLog2<NotificationMetrics, NotificationMetricsClassification>('notification:show', {
+					id: hash(notification.message.toString()),
+					source: notification.source
+				});
+			}
+		}));
+
+		this._register(this.notificationService.onDidRemoveNotification(notification => {
+			if (!notification.silent) {
+				this.telemetryService.publicLog2<NotificationMetrics, NotificationMetricsClassification>('notification:close', {
+					id: hash(notification.message.toString()),
+					source: notification.source
+				});
+			}
+		}));
+	}
+}
```

Comment:
I think our formatting rules still require one final newline

Reply:
There is a newline at the end, GitHub is incorrect here.

Reply:
Ok weird, I use the GH PR extension and it wasn't showing a final new line (cc @alexr00 )

Reply:
@RMacfarlane FYI about the final newline.